app/controllers/Model.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Created by szymon on 5/1/17.
+ */
+public abstract class Model {
+
+    protected Database db;
+
+    final void save() throws NoSuchFieldException, IllegalAccessException {
+        if(db == null) {
+            throw new NullDataBaseExcepion();
+        }
+
+        boolean New = true;
+
+        Connection connection = db.getConnection();
+        String sql;
+        sql = "SELECT id FROM " + this.getClass().getSimpleName().toLowerCase()
+                   + " where id = " + this.getClass().getDeclaredField("id").get(this).toString();
+
+        Statement statement = null;
+
+        try {
+            statement = connection.createStatement();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            ResultSet result = statement.executeQuery(sql);
+            if(result.next()) {
+                New = false;
+            }
+
+            result.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            statement.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            connection.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        if(New) {
+            insert(true);
+        } else {
+            update();
+        }
+    }
+
+    final void get(int id) throws NoSuchFieldException, IllegalAccessException {
+        if(db == null) {
+            throw new NullDataBaseExcepion();
+        }
+
+        Connection connection = db.getConnection();
+        String sql;
+        sql = "SELECT * FROM " + this.getClass().getSimpleName().toLowerCase()
+                + " where id = " + id;
+        Statement statement = null;
+
+        try {
+            statement = connection.createStatement();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            ResultSet result = statement.executeQuery(sql);
+
+            if(!result.next()) {
+                throw new NoSuchElementException();
+            } else {
+                Field[] fields = this.getClass().getDeclaredFields();
+
+                for(Field field : fields) {
+                    field.set(this, result.getObject(makeSql(field.getName())));
+                }
+
+            }
+
+            result.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            statement.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            connection.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    final void insert(boolean New) throws IllegalAccessException, NoSuchFieldException {
+        Connection connection = db.getConnection();
+        String sql;
+        sql = "INSERT INTO " + this.getClass().getSimpleName().toLowerCase() + " VALUES ";
```

Comment:
this.getClass().getSimpleName().toLowerCase() seems to occur many times in code, extract that to separate helper method